spec/src/main/asciidoc/repository.asciidoc

```diff
@@ -371,6 +373,67 @@ Page<User> findByNamePrefix(String namePrefix,
 List<User> findByNamePrefix(String namePrefix, Sort<?>... sorts);
 ----
 
+=== Restrictions
+
+A method annotated `@Find` or `@Delete` may have a <<special-parameters,special parameter>> of type `Restriction<T>`.
+This parameter allows the caller of a repository method to programmatically express additional restrictions on the results returned by the repository method, often using a <<metamodel-java-processor,generated metamodel class>> for type safety.
+
+For example:
+
+[source,java]
+----
+@Find
+List<Product> search(Restriction<Product> restriction);
+
+@Query("WHERE (product.numAvailable > 0)")
+Page<Product> findAvailable(Restriction<Product> restriction,
+                            PageRequest pageRequest,
+                            Order<Product> order);
+----
+
+Programmatic restrictions passed to a parameter of type `Restriction` combine conjunctively with restrictions expressed via the query specified by a `@Query` annotation or via parameters of a method annotated `@Find`.
+
+For example:
+
+[source,java]
+----
+@Repository
+public interface ProductRepository {
+   @Find
+   List<Product> search(@By(_Product.NAME) name,
+                        Restriction<Product> restriction);
+
+   @Find
+   List<Product> search(Restriction<Product> restriction);
```

Comment:
The prior text claimed these to be example of combining with restrictions expressed by `@Query` or `@Find` parameters.  This second example method did neither of those.  No example was included for `@Query`, so I'll recommend switching it to that,

```suggestion
   @Query("WHERE (product.discount > 0)")
   List<Product> discounted(Restriction<Product> restriction);
```

Reply:
Sure, omitting this example method would also work.  That would leave the other method as an example that is consistent what it says it's an example of.